src/main/scala/za/co/absa/spark/partition/sizing/sizer/FromSchemaWithSummariesSizer.scala

```diff
@@ -26,20 +26,23 @@ import za.co.absa.spark.partition.sizing.types.{ByteSize, DataTypeSizes}
 @Experimental
 class FromSchemaWithSummariesSizer(implicit dataTypeSizes: DataTypeSizes) extends RecordSizer {
 
-  override def performRowSizing(df: DataFrame): ByteSize = {
+  override def performRowSizing(df: DataFrame, recordCount: Option[Int] = None): ByteSize = {
```

Comment:
I am not particularly happy with this approach (it's not very elegant), But good enough for now 👍

Reply:
One thing thou - U would consider
```suggestion
  override def performRowSizing(df: DataFrame, dfRecordCount: Option[Int] = None): ByteSize = {
```
or
```suggestion
  override def performRowSizing(df: DataFrame, recordCountInDf: Option[Int] = None): ByteSize = {
```
So it's clear what record count the parameter it is.